test/regression_tests/consensus/consensus_test.tla

```diff
@@ -22,7 +22,12 @@ THEOREM Invariance == Spec => []Inv
 <1>1 Init => Inv
   BY CardinalityZero, SMT DEFS Init, Inv
 <1>2. Inv /\ [Next]_chosen => Inv'
-  BY CardinalityOne, SMT DEFS Next, Inv
+  \* BY CardinalityOne, SMT DEFS Next, Inv
+  <2>1. Inv /\ UNCHANGED chosen => Inv'
+    BY SMT DEF Inv
+  <2>2. Inv /\ Next => Inv'
+    BY CardinalityOne, SMT DEF Next, Inv
```

Comment:
This step doesn't go through on my machine, I had to expand this to:

    <2>2. Inv /\ Next => Inv'
      <3>1. Inv /\ Next => (chosen \subseteq Value)' BY CardinalityOne, SMT DEF Next, Inv 
      <3>2. Inv /\ Next => IsFiniteSet(chosen)' BY CardinalityOne, SMT DEF Next, Inv
      <3>3. Inv /\ Next => (Cardinality(chosen) \leq 1)' BY CardinalityOne, SMT DEF Next, Inv
      <3> QED BY <3>1, <3>2, <3>3 DEF Inv

The conjunction of <3>2 and <3>3 doesn't go through here either.

Reply:
The proof as written by Damien works for me – but I suggest replacing the `Sets' module by the library module `FiniteSetTheorems' and CardinalityZero / CardinalityOne by the corresponding library lemmas.
[consensus_test.txt](https://github.com/tlaplus/tlapm/files/4550306/consensus_test.txt)

Reply:
@muenchnerkindl I have implemented your suggestion and then simplified the proof further. It goes through quite quickly on my machine.